content/mysql/concepts/bit-functions/bit-functions.md

```diff
@@ -0,0 +1,177 @@
+---
+Title: 'Bit Functions'
+Description: 'Commonly used MySQL bit-functions'
+Subjects:
+  - 'Data Science'
+  - 'Computer Science'
+Tags:
+  - 'MySQL'
+  - 'bit functions'
+CatalogContent:
+  - 'getting-started-off-platform-for-data-science'
+  - 'paths/design-databases-with-mysql'
+---
+
+MySQL bit-functions are used to perform operations on binary data at the bit level.
+These functions are used for tasks that require binary data organization or manipulation based on bitwise logic.
+
+## Syntax
+
+These are some commonly used bit-functions in MySQL:
+
+`Bitwise AND &()`: It takes two integer operands and compares them bit by bit.
+The result is 1 if the bits of both operands are 1, otherwise, the result is 0.
+
+```mysql
+SELECT (expression1 & expression2);
+```
+
+`Bitwise XOR ^()`: It takes two integer values and executes a bitwise exclusive OR operation on them.
+
+```psuedo
+SELECT (expression1 ^ expression2);
+```
+
+`Bitwise OR |`: It takes two operands and compares them by the bit. If at least one
+of the corresponding bits is 1, result is 1. If both bits are 0, the result is 0.
+
+```psuedo
+SELECT (expression1 | expression2);
+```
+
+`BIT_COUNT`: This function is used to return the number of bits that are set to 1 in the operand's binary representation.
+
+```psuedo
+SELECT BIT_COUNT(expression);
+```
+
+## Examples
+
+Next are a few examples of some of the MySQL concepts described. In this basic example,
+mySQL performs the bitwise AND operation on the two literal integer values 45 and 25:
+
+```mysql
+> SELECT 45 & 25;
+-> 9;
+```
+
+In the above example:
+
+- 45 in binary is represented as 101101.
+- 25 in binary is represented as 011001.
+- The bitwise AND operation between 45 and 25 is in binary 001001.
+- 001001 in decimal is represented as 9.
+
+In this example, the function `XOR` is used to compare and manipulate the label values within a database.
+The query uses this function to select rows based on the results of the operation:
+
+```mysql
+SELECT *
+  FROM posts
+  WHERE (labels ^ 3) > 5;
+```
+
+From the table, a query takes rows from the name where the XOR comparison of labels with 3 results in a value greater than 5.
+id|name |labels
+--|------|-----
+1 | post1| 5
+| |
+2 | post2| 3
+| |
+3 | post3| 12
+
+Performed XOR operations are explained bellow:
+
+Labels value 5 is represented by 0101 and XOR comparison is:
+
+- 0101 ^ 0011 = 0110 (6 in decimal, which is greater than 5)
+
+Labels value 3 is represented by 0011 and XOR comparison is:
+
+- 0011 ^ 0011 = 0000 (0 in decimal, which is not greater than 5)
+
+Labels value 12 is represented by 1100 and XOR comparison is:
+
+- 1100 ^ 0011 = 1111 (15 in decimal, which is greater than 5)
```

Comment:
I'm afraid this will trigger MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- 0011 ^ 0011 = 0000 (0 in dec..."] (& more)..